### ISSUE TITLE
Mac Catalyst support missing. Please add support!

### ISSUE THREAD

**Issue by parmar-mehul** [Tencent/MMKV#1345]
### The language of MMKV
Swift

### The version of MMKV
v1.3.7 LTS

### The platform of MMKV
iOS, Specifically Mac Catalyst

### The installation of MMKV
Git clone

### IDE
Apple M2 Pro MacOS Sonoma 14.5 (23F79), XCode Version 15.4 (15F31d)

### What's the issue?
While creating archive for "Any Mac, Mac Catalyst", Getting compile error, 
Problem is, Unable to generate archive. Blocker issue.
Attached screenshot as suggested. 

![Screenshot 2024-07-09 at 11 41 06 AM](https://github.com/Tencent/MMKV/assets/98015131/aae51d16-da1b-479e-a812-c7291ada3087)

Kindly help and release new version.

**Comment by lingol** [comment]
Noted. I have never thought Mac Catalyst be used.

**Comment by parmar-mehul** [comment]
@lingol 
Please inform us in this same thread once you add support, Thank you very much for your valuable time! 

_(We are waiting for new release!)_

**Comment by lingol** [comment]
A few tweaks and it runs on Mac Catalyst now. You can try it out in the dev branch.

**Comment by parmar-mehul** [comment]
Hello @lingol, It works tested & confirm. 

Thank you for your quick support! Let's move this https://github.com/Tencent/MMKV/commit/32ed8ce54488cbc1828d390a4eac7cb94c9f22ad changes from dev to master.

---

FYI, I have also created a XCFramework from master branch, also added spm (swift package manager) support as well,
- https://github.com/parmar-mehul/MMKV-XCFramework

FYI, I have created a XCFramework from dev branch attached here. 
- checksum: f48f67bcbd431ef5e1dbfe3536aa866fceb5c99cda2b6d215b3c6ceba17d56ec
- xcframework: [MMKV.xcframework.zip](https://github.com/user-attachments/files/16158837/MMKV.xcframework.zip)
